crates/client/src/test_generator.rs

```diff
@@ -212,18 +212,26 @@ impl TestGenerator {
         let fuzz_id = if fuzz_dir_path.read_dir()?.next().is_none() {
             0
         } else {
-            let mut directories: Vec<_> = fuzz_dir_path
+            let mut directories: std::collections::HashSet<_> = fuzz_dir_path
```

Comment:
Looks good. Just try to avoid unwraps if possible. And the loop could in theory (not very likely though) result in infinite loop.

What about something like that:
```rust
let mut directories: std::collections::HashSet<_> = fuzz_dir_path
                .read_dir()
                .expect("read_dir call failed")
                .map(|r| {
                    r.expect("read_dir DirEntry error")
                        .file_name()
                        .to_string_lossy()
                        .to_string()
                })
                .collect();

            // INFO discard known entries created by framework, everything else
            // created by user will be taken as fuzz test.
            directories.retain(|x| x != "fuzzing");
            directories.retain(|x| x != "Cargo.toml");

            let mut fuzz_id = directories.len();
            loop {
                let fuzz_test = format!("fuzz_{fuzz_id}");
                if directories.contains(&fuzz_test) && fuzz_id < usize::MAX {
                    fuzz_id += 1;
                } else {
                    break fuzz_id;
                }
            }
```

Reply:
thanks for feedback, will update